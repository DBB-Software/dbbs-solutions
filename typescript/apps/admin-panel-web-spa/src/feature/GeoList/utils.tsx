import React from 'react'
import { GridColDef, GridRenderCellParams } from '@dbbs/mui-components'
import moment from 'moment-timezone'
import { Link } from '@tanstack/react-router'

export const getGeoColumnModel = (): GridColDef[] => [
  { field: 'id', headerName: 'ID', editable: false },
  { field: 'bubbleId', headerName: 'Bubble ID', width: 250, editable: false },
  { field: 'name', headerName: 'Name', width: 250, editable: false },
  {
    field: 'geoType',
    headerName: 'Geo Type',
    width: 150,
    editable: false
  },
  {
    field: 'timezone',
    headerName: 'Time Zone',
    width: 200,
    type: 'singleSelect',
    valueOptions: moment.tz.names(),
    editable: false
  },
  { field: 'createdBy', headerName: 'Created By', width: 250, editable: false },
  {
    field: 'createdDate',
    headerName: 'Created Date',
    type: 'dateTime',
    valueGetter: (value: string) => new Date(value),
    width: 200,
    editable: false
  },
  {
    field: 'modifiedDate',
    headerName: 'Modified Date',
    type: 'dateTime',
    valueGetter: (value: string) => new Date(value),
    width: 200,
    editable: false
  },
  { field: 'scheduleUrl', headerName: 'Schedule URL', width: 350, editable: false },
  {
    field: 'captureScheduleFlag',
    headerName: 'Capture Schedule',
    type: 'boolean',
    width: 150,
    editable: false
  },
  { field: 'captureStreamFlag', headerName: 'Capture Stream', type: 'boolean', width: 150, editable: false },
  { field: 'scheduleFormat', headerName: 'Schedule Format', width: 150, editable: false },
  { field: 'streamType', headerName: 'Stream Type', width: 150, editable: false },
  {
    field: 'jurisdiction',
    headerName: 'Jurisdiction',
    width: 100,
    type: 'boolean',
    editable: false
  },
  {
    editable: false,
    field: 'parent',
    headerName: 'Parent',
    width: 300,
    renderCell: (params: GridRenderCellParams) => (
      <Link to={`/geo/${params.row?.parent?.id}/edit`} target="_blank">
        {params.row?.parent?.fullName || params.row?.parent?.name || ''}
      </Link>
    )
  },
  { field: 'sortKey', headerName: 'Sort Key', width: 300, editable: false },
  { field: 'channelUrl', headerName: 'Channel URL', width: 300, editable: false },
  { field: 'statusSchedule', headerName: 'Schedule Status', width: 150, editable: false },
  { field: 'mirrorGeo', headerName: 'Mirror Geo ID', width: 300, editable: false },
  { field: 'bubbleId', headerName: 'Bubble ID', width: 150, editable: false },
  { field: 'geoParentId', headerName: 'Geo Parent ID', width: 150, editable: false },
  { field: 'detectStartMethod', headerName: 'Detect Start Method', width: 200, editable: false },
  { field: 'detectEndMethod', headerName: 'Detect End Method', width: 200, editable: false },
  { field: 'statusStream', headerName: 'Status Stream', width: 150, editable: false },
  { field: 'flagOnlyAgenda', headerName: 'Flag Only Agenda', type: 'boolean', width: 150, editable: false },
  { field: 'flagOptInOnly', headerName: 'Flag Opt-In Only', type: 'boolean', width: 150, editable: false },
  { field: 'singlePlayerUrl', headerName: 'Single Player URL', width: 300, editable: false },
  { field: 'flagLive', headerName: 'Flag Live', type: 'boolean', width: 150, editable: false },
  { field: 'detectEndOcrString', headerName: 'Detect End OCR String', width: 200, editable: false },
  { field: 'debug', headerName: 'Debug', type: 'boolean', width: 100, editable: false },
  { field: 'demo', headerName: 'Demo', type: 'boolean', width: 100, editable: false },
  { field: 'scheduleRefreshFrequency', headerName: 'Schedule Refresh Frequency', width: 200, editable: false }
]
