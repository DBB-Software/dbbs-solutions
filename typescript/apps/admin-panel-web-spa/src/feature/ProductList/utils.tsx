import { GridColDef, GridRenderCellParams } from '@dbbs/mui-components'

export const getProductColumnModel = (): GridColDef[] => [
  { field: 'id', headerName: 'ID', width: 100, editable: false },
  { field: 'typeId', headerName: 'Type ID', width: 150, editable: false },
  { field: 'name', headerName: 'Name', width: 250, editable: false },
  { field: 'description', headerName: 'Description', width: 300, editable: false },
  { field: 'price', headerName: 'Price', width: 150, type: 'number', editable: false },
  { field: 'currency', headerName: 'Currency', width: 100, editable: false },
  { field: 'sku', headerName: 'SKU', width: 150, editable: false },
  { field: 'inventoryCount', headerName: 'Inventory Count', width: 150, type: 'number', editable: false },
  { field: 'imageUrl', headerName: 'Image URL', width: 300, editable: false },
  { field: 'isActive', headerName: 'Is Active', width: 150, type: 'boolean', editable: false },
  {
    field: 'createdAt',
    headerName: 'Created At',
    type: 'dateTime',
    valueGetter: (value) => new Date(value),
    width: 200,
    editable: false
  },
  {
    field: 'updatedAt',
    headerName: 'Updated At',
    type: 'dateTime',
    valueGetter: (value) => new Date(value),
    width: 200,
    editable: false
  },
  {
    field: 'type',
    headerName: 'Type',
    width: 300,
    renderCell: (params: GridRenderCellParams) => params.row?.type?.name || '',
    editable: false
  }
]
