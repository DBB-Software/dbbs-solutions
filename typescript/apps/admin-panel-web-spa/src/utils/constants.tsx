import React, { ReactNode } from 'react'
import { MapPin, Users } from 'lucide-react'
import { GridColDef } from '@dbbs/mui-components'

export const BASE_EDITABLE_FIELD: Partial<GridColDef> = {
  width: 150,
  editable: true
}

export type NavigationLink = {
  label: string
  key: string
  link: string
  icon?: ReactNode
}

export const DEFAULT_CONFIG_PAGE_GRID_SIZE = { xs: 12, md: 6 }

export const SIDEBAR_DATA_PAGES_LINKS: NavigationLink[] = [
  {
    label: 'Geo',
    link: '/geo',
    key: 'geo',
    icon: <MapPin />
  },
  {
    label: 'Meeting Type',
    link: '/meeting-type',
    key: 'meetingType',
    icon: <Users />
  }
]
