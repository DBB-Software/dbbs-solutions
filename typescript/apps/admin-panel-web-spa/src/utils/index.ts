export * from './buildQueryParams'
export * from './constants'
export * from './makeSxStyles'
export * from './buildListQuery'
export * from './displayValidationErrors'
export * from './renderDefaultActions'
