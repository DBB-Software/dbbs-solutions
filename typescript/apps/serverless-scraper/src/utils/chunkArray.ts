export function chunkArray<T>(array: T[], chunkSize: number): T[][] {
  if (chunkSize <= 0) {
    throw new Error('Chunk size must be a positive number')
  }

  const chunks: T[][] = []
  for (let i = 0; i < array.length; i += chunkSize) {
    chunks.push(array.slice(i, i + chunkSize))
  }
  return chunks
}
