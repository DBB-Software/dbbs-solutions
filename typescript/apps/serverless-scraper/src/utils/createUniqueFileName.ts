export const createUniqueFileName = (value: string, filetype: string) => {
  const timestamp = new Date().toISOString().replace(/[-:.]/g, '')

  return `${timestamp}_${value.replaceAll(' ', '_')}.${filetype}`
}
