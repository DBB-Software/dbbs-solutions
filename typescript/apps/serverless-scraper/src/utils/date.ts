import moment from 'moment/moment.js'

export const getYesterdayDateString = (stringFormat = 'YYYY-MM-DD') => moment().subtract(1, 'days').format(stringFormat)
