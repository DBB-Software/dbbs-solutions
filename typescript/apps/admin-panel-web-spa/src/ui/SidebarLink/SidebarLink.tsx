import React, { FC } from 'react'
import { Link } from '@tanstack/react-router'
import { ListItemButton, ListItemIcon, Typography } from '@mui/material'
import { makeSxStyles, NavigationLink } from '../../utils'
import theme from '../../theme'

export type SidebarLinkProps = NavigationLink & { isActive: boolean }

const makeStyles = () =>
  makeSxStyles({
    listIcon: {
      minWidth: 30,
      svg: {
        width: 20
      },
      color: 'inherit'
    },
    sidebarLink: {
      fontSize: 14
    },
    iconWrapper: {
      display: 'flex',
      position: 'relative',
      alignItems: 'center',
      justifyContent: 'center'
    }
  })

export const SidebarLink: FC<SidebarLinkProps> = ({ link, icon, label, isActive }) => {
  const styles = makeStyles()

  return (
    <Link
      style={{
        textDecoration: 'none',
        color: 'inherit'
      }}
      to={link}
    >
      <ListItemButton
        sx={{
          backgroundColor: isActive ? theme.palette.secondary.light : undefined
        }}
      >
        <ListItemIcon sx={styles.listIcon}>{icon}</ListItemIcon>
        <Typography sx={styles.sidebarLink}>{label}</Typography>
      </ListItemButton>
    </Link>
  )
}
