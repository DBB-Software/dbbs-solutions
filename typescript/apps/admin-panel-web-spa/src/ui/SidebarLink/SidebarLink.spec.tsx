import { render } from '@testing-library/react'
import React from 'react'
import { SidebarLink } from './SidebarLink'
import { NavigationLink } from '../../utils'

const mockLink: NavigationLink & { isActive: boolean } = {
  label: 'Dashboard',
  link: '/dashboard',
  key: 'dashboard',
  isActive: false
}

jest.mock('@tanstack/react-router', () => ({
  ...jest.requireActual('@tanstack/react-router'),
  useRouterState: jest.fn(() => ({
    location: { pathname: '/' }
  })),
  Link: jest.fn(({ to, children }) => <a href={to}>{children}</a>)
}))

jest.mock('@dbbs/mui-components', () => ({
  ...jest.requireActual('@dbbs/mui-components'),
  useMediaQuery: jest.fn(() => true)
}))

describe('<Layout />', () => {
  it('renders correctly', async () => {
    const { asFragment } = render(<SidebarLink {...mockLink} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
