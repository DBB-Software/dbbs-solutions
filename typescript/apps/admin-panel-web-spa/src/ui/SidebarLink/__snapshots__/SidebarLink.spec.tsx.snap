// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders correctly 1`] = `
<DocumentFragment>
  <a
    href="/dashboard"
  >
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root css-1a5375a-MuiListItemIcon-root"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-1d9ye53-MuiTypography-root"
      >
        Dashboard
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </a>
</DocumentFragment>
`;
